### PR TITLE
Add -indirect flag to rdeps which prints indirect dependencies.

### LINKED ISSUE #42: rdeps: add a flag to print indirect dependencies
We would like to use `rdeps` for limiting the amount of test one needs to run on pre-push hook. If one modifies a package X, we want to run tests for X and for all packages that depend on X.

Currently, with dependency tree `X <= Y <= Z`, `rdeps X` yields `Y`, and there is no easy way to get both `Y` and `Z`. We would like to implement a `-indirect` (or `-recursive`?) flag, that will result in `rdeps -indirect X` returning both `Y` and `Z`.

I'll happy to implement this and open a pull request, I just want to hear your opinion on that.

### PR DESCRIPTION
**Pull request by mwek**
Closes #42

### CODE REVIEW

File: cmd/rdeps/rdeps.go [hunk 1]
```diff
@@ -22,6 +21,7 @@ func main() {
 	var tags buildutil.TagsFlag
 	flag.Var(&tags, "tags", "List of build tags")
 	stdin := flag.Bool("stdin", false, "Read packages from stdin instead of the command line")
+	indirect := flag.Bool("indirect", false, "Print also indirect dependencies")
```
Comment (dominikh) [inline cmd/rdeps/rdeps.go]: Call the flag `r` (for recursive), with the description `Print reverse dependencies recursively`. Call the variable `recursive`.

File: cmd/rdeps/rdeps.go [hunk 2]
```diff
@@ -53,11 +53,24 @@ func main() {
 	}
 	_, reverse, errors := importgraph.Build(&ctx)
 	_ = errors
-	for _, pkg := range pkgs {
+
+	printed := map[string]bool{}
```
Comment (dominikh) [inline cmd/rdeps/rdeps.go]: Call the variable `seen`

File: cmd/rdeps/rdeps.go [hunk 3]
```diff
@@ -53,11 +53,24 @@ func main() {
...
+	var printRDeps func(pkg string)
+	printRDeps = func(pkg string) {
 		for rdep := range reverse[pkg] {
-			fmt.Println(rdep)
+			if !printed[rdep] {
```
Comment (dominikh) [inline cmd/rdeps/rdeps.go]: `if printed[rdep] { continue }`, then you don't need to indent the common case.

**Comment by mwek** [comment]
All comments are addressed, I believe this is ready to be merged.

**Comment by dominikh** [comment]
Merged, thanks.

**Pull request closed without merging**